be/dtm0_log.h

```diff
@@ -0,0 +1,235 @@
+ *  M0_DTPS_PERSISTENT. On originator the state of the distributed transaction
+ *  (dtx) can be M0_DTPS_INPROGRESS to M0_DTPS_PERSISTENT for each participant.
+ *
+ *  During recovery operation of any of the participant, rest of the participant
+ *  and originator will iterate over the logged journal and extract the state
+ *  of each transation for participant under recovery from logged information
+ *  and will send redo request for those distributed transaction(dtx) which  are
+ *  not M0_DTPS_PERSISTENT on participant being recovered.
+ *
+ *  Upon receiving redo request participant under recovery will log the state
+ *  of same distributed transaction(dtx) of remote participant in persistent
+ *  store.
+ *
+ * @section Usecases
+ *
+ * 1. There is a dedicated function m0_be_dtm0_log_create which will create
+ *    m0_be_dtm0_log during mkfs procedure.
+ *
+ * 2. When distributed transaction(dtx) successfully executed on participant,
+ *    DTM0 log record will be added by participant to persistent store and where
+ *    participant state in group of state will be M0_DTPS_EXECUTED and for
+ *    rest of the participant state will be logged as it is.
+ *
+ * 3. When distributed transaction(dtx) become persistent on particular
+ *    participant, the state of the distributed transaction(dtx) for this
+ *    participant will be updated as M0_DTPS_PERSISTENT.
+ *
+ * 4. When distributed transaction(dtx) become persistent on remote participant,
+ *    the participant sends the persistent notice to rest of the participants
+ *    and originator stating that distributed transaction(dtx) is persistent on
+ *    its store. Upon receiving persistent notice each of the participants and
+ *    originator will update the state of distributed transaction(dtx) for
+ *    remote participant as M0_DTPS_PERSISTENT in DTM0 log record.
+ *
+ *    There is one special case where a process got a persistent notice from
+ *    another process but the corresponding request and reply were not
+ *    received/processed yet. In this case, the DTM service should add a special
+ *    log record: a log record with empty payload (NULL). The users of DTM0 log
+ *    should be ready to encounter such a log entry, and treat it differently
+ *    whenever it is required.
+ */
+
+enum m0_be_dtm0_log_credit_op {
+	M0_DTML_CREATE,		/**< m0_be_dtm0_log_create() */
+	M0_DTML_SENT,		/**< m0_be_dtm0_log_update() */
+	M0_DTML_EXECUTED,	/**< m0_be_dtm0_log_update() */
+	M0_DTML_PERSISTENT,	/**< m0_be_dtm0_log_update() */
+	M0_DTML_PRUNE,		/**< m0_be_dtm0_log_prune()  */
+	M0_DTML_REDO            /**< m0_be_dtm0_log_update() */
+};
+
+struct m0_dtm0_log_rec {
+	struct m0_dtm0_dtx     dlr_dtx;
+	struct m0_dtm0_tx_desc dlr_txd;
+	struct m0_be_list_link dlr_link;    /**< link into m0_be_dtm0_log::tlist */
+	uint64_t               dlr_magic;
+	struct m0_tlink        dlr_tlink;
+	struct m0_buf          dlr_payload;
+};
+
+struct m0_be_dtm0_log {
```

Comment:
Please add Doxygen comments for this new data struct.
Please also mention that this is in-memory data struct.

Reply:
Done.